fix(overflow-menu): updated aria-label and roles, made ul a sibling

harpality opened the pull request:
addresses #2495

Updated a11y issues as requested.

#### Changelog

**New**

- line 551 added `{menuBody}` so that `<ul role=‘menu’>` is a sibling to `<div role=“button”>`. wrapped return in `<React.Fragment>`.
- line 500 added `aria-label=“Menu”` to`ul` for `menuBody` 

**Changed**

- line 534 changed `role=“menu”` to `role=“button”`
- line 541 changed `aria-label` from `{ariaLabel}` to `“Menu”`

**Removed**

- line 502 removed extra `<div role=“menuitem”>`

harpality committed:
fix(overflow-menu): updated aria-label and roles, made ul a sibling

harpality committed:
fix(overflow-menu): removing unused var

harpality committed:
fix(overflow-menu): update snapshots

asudoh commented on packages/react/src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -525,26 +526,29 @@ class OverflowMenu extends Component {
+            onKeyDown={this.handleKeyPress}
+            onBlur={this.handleBlur}
+            onClick={this.handleClick}
+            aria-label="Menu"
```
Instead of killing `ariaLabel`, do you want to change `defaultProps` to make `Menu` the default?

asudoh commented on packages/react/src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -493,27 +492,29 @@ class OverflowMenu extends Component {
+        className={overflowMenuOptionsClasses}
+        tabIndex="-1"
+        role="menu"
+        aria-label="Menu">
```
Do you want to make this refer to `ariaLabel` prop?

asudoh reviewed (commented):
Thanks @harpalassi for jumping in!

joshblack committed:
chore(project): use yarn policies to set version to 1.16.0 (#2902)

* chore(project): use yarn policies to set version to 1.16.0

* chore(project): update prettier to ignore .yarn folder

asudoh committed:
fix(modal): support different numbers of buttons (#2900)

Fixes #2325.

joshblack committed:
chore(examples): build examples based on whitelist (#2903)

dakahn commented on packages/react/src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -493,27 +492,29 @@ class OverflowMenu extends Component {
+        className={overflowMenuOptionsClasses}
+        tabIndex="-1"
+        role="menu"
+        aria-label="Menu">
```
```suggestion
        aria-label={ariaLabel}>
```

harpality replied:
`{ariaLabel}` re-added, `Menu` added to `defaultProps` on line 209 and line 23 in `OverflowMenu-story.js`

dakahn commented on packages/react/src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -440,7 +440,6 @@ class OverflowMenu extends Component {
     const {
       id,
       tabIndex,
```
```suggestion
      tabIndex,
      ariaLabel,
```

harpality replied:
Changes made - ariaLabel re-added.

dakahn reviewed (changes requested):
Great work @harpalassi. Thanks for the contrib. Typically we want to modify a component's API (all the props and various configs someone using the component might set) as little as possible unless we're dealing with a major release -- where we can make breaking changes. So yeah, making the `defaultProp` be `menu` is the way to go. 👍🏽

joshblack committed:
chore(release): v10.3.0-rc.2 (#2908)

harpality committed:
fix(overflow-menu): update defaultprops for Menu

harpality committed:
fix(overflow-menu): removed add'l this.props

asudoh reviewed (approved):
LGTM 👍 - Thanks @harpalassi!

asudoh commented:
BTW would you update the snapshot (again)...? Thanks!

harpality committed:
fix(overflow-menu): update snapshots

dakahn reviewed (approved):
🏄 🏄 🏄 Way to go

asudoh committed:
fix(radio-button): group-level label/button alignment (#2914)

Fixes #2286.

emyarod commented on packages/react/src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -493,27 +493,29 @@ class OverflowMenu extends Component {
+        className={overflowMenuOptionsClasses}
+        tabIndex="-1"
+        role="menu"
+        aria-label={ariaLabel}>
```
```suggestion
        aria-label={ariaLabel || this.props['aria-label']}>
```

should we also pass in the standard aria attribute as an option?

emyarod commented on packages/react/src/components/OverflowMenu/OverflowMenu.js:
```diff
@@ -525,26 +527,29 @@ class OverflowMenu extends Component {
+            {open && wrappedMenuBody}
+          </div>
+        </ClickListener>
+        {menuBody}
```
the menu body is being rendered twice, once in a `<FloatingMenu>` (`wrappedMenuBody` on L548) and once without (`menuBody` on L551). but I don't think we can simply move the menu body and eliminate the floating menu wrapper without losing functionality like keyboard navigation

emyarod reviewed (changes requested):
![image](https://user-images.githubusercontent.com/8265238/58884884-d9e70580-86a6-11e9-8a28-3ed9925cbb0f.png)

there's a visual bug where the overflow menu overlaps with the trigger button. this might be because the menu body is being rendered in two separate locations. this could also be affecting keyboard navigation

the aria attribute changes look good to me, but the fix for the screen reader getting lost (making the `ul[role="menu"]` a sibling of the `div[role="button"]`) is not so straightforward as replacing `wrappedMenuBody` with `menuBody` and moving it. instead I think we should modify the target of the `<FloatingMenu>` portal

we can probably split the fixes into a few PRs and scope this PR to just the aria label, `role`, and `div.menuitem` changes? or would you want to tackle the floating menu changes as well @harpalassi ?

Commit:
Fix sass error (#2905)

dependabot[bot] committed:
chore(deps): bump sshpk in /packages/react/examples/create-react-app (#2915)

Bumps [sshpk](https://github.com/joyent/node-sshpk) from 1.13.1 to 1.16.1.
- [Release notes](https://github.com/joyent/node-sshpk/releases)
- [Commits](https://github.com/joyent/node-sshpk/compare/v1.13.1...v1.16.1)

dakahn commented:
@emyarod yeah, let's split out that work, i'll add your findings to the and we can get another PR going for those larger floating menu changes. @harpalassi can you rollback your changes to the `ul`?

harpality committed:
fix(overflow-menu): rollback ul changes

harpality commented:
@dakahn sounds good - i just rolled back the changes to `ul`.

emyarod committed:
fix(tooltip): set width to content max (#2923)

emyarod reviewed (approved):
looks good to me, I guess the `ariaLabel` prop is sufficient then

dependabot[bot] committed:
chore(deps): bump safer-eval (#2930)

Bumps [safer-eval](https://github.com/commenthol/safer-eval) from 1.2.3 to 1.3.3.
- [Release notes](https://github.com/commenthol/safer-eval/releases)
- [Commits](https://github.com/commenthol/safer-eval/compare/v1.2.3...v1.3.3)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump safer-eval in /packages/colors/examples/preview (#2931)

Bumps [safer-eval](https://github.com/commenthol/safer-eval) from 1.2.3 to 1.3.3.
- [Release notes](https://github.com/commenthol/safer-eval/releases)
- [Commits](https://github.com/commenthol/safer-eval/compare/v1.2.3...v1.3.3)

Signed-off-by: dependabot[bot] <support@github.com>

asudoh committed:
fix(button): require iconDescription for icon-only (#2929)

* fix(Button): require iconDescription for icon-only

Fixes #2884.

* chore(test): update snapshots

emyarod commented:
snapshots need to be updated once again because of the latest changes

harpality committed:
fix(overflow-menu): update snapshots

harpality committed:
Revert "fix(overflow-menu): update snapshots"

This reverts commit 1015db64365b6c06a08e740a185690ed11fee7f1.

harpality committed:
fix(overflow-menu): update snapshots

harpality committed:
`Revert "fix(overflow-menu): update snapshots"

This reverts commit 6fe35a66877fb872927fb03fc77aadc860e8a53f.

harpality committed:
fix(overflow-menu): update snapshots

joshblack committed:
Merge branch 'master' of github.com:carbon-design-system/carbon into overflowmenu

joshblack committed:
chore(react): remove redundant version field

asudoh commented:
The diff now seems to contain lots of relevant changes - @harpalassi Could you fix? Thanks!

joshblack commented:
@asudoh just pushed up a merge 👍

joshblack committed:
Merge branch 'master' into overflowmenu

joshblack committed:
Merge branch 'master' into overflowmenu

joshblack committed:
Merge branch 'master' into overflowmenu
